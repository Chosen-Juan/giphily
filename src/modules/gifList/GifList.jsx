import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import Gif from './Gif';
import './styles.scss';
import { addGifToCollection } from '../collection/CollectionActions';

export class GifList extends Component {

  static propTypes = {
    gifList: PropTypes.arrayOf(Gif.WrappedComponent.propTypes.gif),
    addGifToCollectionDispatcher: PropTypes.func.isRequired
  };

  static defaultProps = {
    gifList: []
  };

  constructor(props) {
    super(props);
    this.renderGifs = this.renderGifs.bind(this);
  }

  renderGifs() {
    const { addGifToCollectionDispatcher } = this.props;
    return this.props.gifList.map(gif => {
      return <div key={gif.id} className='gif-button' onClick={() => addGifToCollectionDispatcher(gif)}>
        <Gif gif={gif} />
      </div>;
    });
  }

  render() {
    return (
      <div className='gif-container'>
        {this.renderGifs()}
      </div>
    );
  }
};

const mapStateToProps = state => ({
  gifList: state.searchReducer.items
});

const mapDispatchToProps = dispatch => ({
  addGifToCollectionDispatcher: gif => dispatch(addGifToCollection(gif))
});

export default connect(mapStateToProps, mapDispatchToProps)(GifList);
