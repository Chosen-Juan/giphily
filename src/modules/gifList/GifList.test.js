import React from 'react';
import { shallow } from 'enzyme';

import { GifList } from './GifList';

const addGifSpy = jest.fn();
const image = {
  url: '://fakegif.gif'
};
const defaultProps = {
  addGifToCollectionDispatcher: addGifSpy,
  gifList: [{
    title: 'Fake gify',
    id: 'abc',
    images: {
      original: image,
      original_still: image
    }
  }]
};


describe('GifList', () => {
  let gifList;

  beforeEach(() => {
    gifList = shallow(<GifList {...defaultProps} />);
  });

  it('should match snapshot', () => {
    expect(gifList).toMatchSnapshot();
  });

  it('should call onClick with the gif info', () => {
    const div = gifList.find('.gif-button');
    div.simulate('click');
    expect(addGifSpy).toBeCalledWith(defaultProps.gifList[0]);
  });

});
