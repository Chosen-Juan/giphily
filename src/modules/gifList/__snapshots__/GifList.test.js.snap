// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifList should match snapshot 1`] = `
<div
  className="gif-container"
>
  <div
    className="gif-button"
    key="abc"
    onClick={[Function]}
  >
    <Connect(Gif)
      gif={
        Object {
          "id": "abc",
          "images": Object {
            "original": Object {
              "url": "://fakegif.gif",
            },
            "original_still": Object {
              "url": "://fakegif.gif",
            },
          },
          "title": "Fake gify",
        }
      }
    />
  </div>
</div>
`;
